### ISSUE TITLE
Can you add your demo code to the repo?

### ISSUE THREAD

**Issue by delebash** [colkassad/terrain-rgb-converter#5]
This is exactly what I am looking for. The way your demo works is the same thing I need for my app. It would be wonderful if you would share the demo code. TY

I added the same issue on your other conversion to but this is the one I am most interested in. TY

Just some info on what I am trying to do.  I am new to mapbox and have been working to create this type of program including your demo app, very cool.  Ultimately, I want to integrate this into Unreal and have it update and scale the landscape when you click on a new area.  I found this article that I think will allow us to update the landscape dynamically instead of manually importing the heightmap.  See https://hippowombat.tumblr.com/post/180615213251/blueprint-powered-landscape-edits-in-ue4-420 and https://blueprintue.com/blueprint/ju_x2bhg/.  Thanks again.

Dan

**Comment by colkassad** [comment]
Note: closing the other issue and moving my answer here.

Looks like my demo code is using an older version where the slope and height were in one file. I'll sort that out, sanitize some stuff, and post the demo repo as a public project as soon as I can. It's an express map running on Node.js. If you just want to see how to use the code in an express route, this is the code for what happens when the user double clicks a tile and the 16bit PNG is downloaded:

```
var express = require('express');
var router = express.Router();
var fs = require('fs');
var converter = require('terrain-rgb-height');

/* GET elevation. */
router.get('/:zoom/:x/:y/:token', function(req, res) {
    var zoom = req.params.zoom;
    var x = req.params.x;
    var y = req.params.y;
  
    var url = 'https://api.mapbox.com/v4/mapbox.terrain-rgb/{z}/{x}/{y}@2x.pngraw?access_token=' + req.params.token;
    url = url.replace('{z}', zoom);
    url = url.replace('{x}', x);
    url = url.replace('{y}', y);
    var inputPng = "./public/images/" + req.params.zoom + "-" + req.params.x + "-" + req.params.y + ".png";
    var outputPng = "./public/images/" + req.params.zoom + "-" + req.params.x + "-" + req.params.y + "-sixteen-bit.png";
    var options = {
      inputFilePath: inputPng,
      outputFilePath: outputPng
    };
	//if the tile was already converted and cached on the server, use that and don't bother converting
    if (fs.existsSync(inputPng)) {
      if (fs.existsSync(outputPng)) {
        res.download(outputPng);
      } else {
        converter.convert(options, function() {
          res.download(outputPng);
        });
      }
    } else {
      converter.getTile(url, inputPng, function() {
        converter.convert(options, function() {
          res.download(outputPng);
        });
      });
    }
  });

module.exports = router;
```

Basically, I created a route that accepts a URL which contains the zoom, x, y, and token (you will need to create a MapBox token if you have not yet), derived from the selected tile in my MapBox map. If that tile has already been selected and converted (I keep them cached on the server using a filename derived from the URL parameters), I serve that to the user. Otherwise I reach out to MapBox, get the tile, and convert/cache it. If you have any questions, let me know. I'll get the repo up as soon as I can.

Glad you find it useful.

**Comment by colkassad** [comment]
Here is the demo map repo: https://github.com/colkassad/terrain-rgb-demo-map

I have not tested this yet. You may need to modify relative paths in some URLs depending on how you run it (locally, on a server, etc). You will also need to update the mapbox api token [here](https://github.com/colkassad/terrain-rgb-demo-map/blob/b37b905b3566db68c935f10ce3b10db7d8644e81/terrain-rgb-demo/public/javascripts/map.js#L3).

**Comment by delebash** [comment]
Oh, wow that was fast, thanks!

**Comment by colkassad** [comment]
No problem! If you have any issues running the demo, feel free to reach out on that repo. I'll do my best to help but I'm not an express expert, unfortunately.

**Comment by delebash** [comment]
ok TY

**Comment by delebash** [comment]
Do you mind sharing your mapbox map?  Just open up your map in mapbox studio > select share from the top right > click on make public botton and send the link.

<img width="960" alt="msedge_nlTX06nnH6" src="https://user-images.githubusercontent.com/2220160/132104311-22068620-cae4-4c4c-951a-fca21788757d.png">

**Comment by colkassad** [comment]
I checked my account and I don't have a map that correspond to my demo map. I seem to remember just copying the style from somewhere:

```
var map = new mapboxgl.Map({
  container: 'map',
  style: 'mapbox://styles/mapbox/cjaudgl840gn32rnrepcb9b9g',
  center: [86.922623, 27.986065], //Mt Everest
  zoom: 9,
  doubleClickZoom: false
});
```

**Comment by delebash** [comment]
TY.  I figured it out.
